lib/elixir/lib/option_parser.ex

```diff
@@ -112,6 +113,11 @@ defmodule OptionParser do
       iex> OptionParser.parse(["--no-op", "path/to/file"], switches: [op: :boolean])
       {[op: false], ["path/to/file"], []}
 
+  ### Parsing undefined switches
+
+  By-default, only switches with defined atoms will be parsed. If you want to dynamically add
```

Comment:
"By-default" => "By default". I believe we should include an example of cases that don't work without the `:allow_inexistent_atoms` option and how they are fixed by adding the option. Basically the case reported in the issue.